ArduPlane/commands_logic.cpp

```diff
@@ -594,22 +594,8 @@ bool Plane::verify_takeoff()
     }
 
     // check for optional takeoff timeout
-    if (takeoff_state.start_time_ms != 0 && g2.takeoff_timeout > 0) {
-        const float ground_speed = gps.ground_speed();
-        const float takeoff_min_ground_speed = 4;
-        if (!arming.is_armed_and_safety_off()) {
-            return false;
```

Comment:
I didn't spot this early return, sorry. You will have to keep this bit. 
```
    if (takeoff_state.start_time_ms != 0 && g2.takeoff_timeout > 0) {
        if (!arming.is_armed_and_safety_off()) {
            return false;
        }
        if (plane.check_takeoff_timeout()) {
            mission.reset();
        }
    }
```
Although I'm fairly sure the early return is not actually doing anything for us. This would probably behave the same but would need more testing:
```
    if (plane.check_takeoff_timeout()) {
        mission.reset();
    }
    if (!arming.is_armed_and_safety_off()) {
        return false;
    }
```

Reply:
I did not think it was really needed...the whole point of the TKOFF_TIMER is to disarm if you have not taken off....like holding it and not releasing, or the safety is on and prop does not spin (actually how I SITL test this...disable the motor on mode takeoff), without having to fumble on the TX or GCS to disarm